recipes/dd-opentracing-cpp/all/conanfile.py

```diff
@@ -22,99 +29,80 @@ class DatadogOpenTracingConan(ConanFile):
+            self.output.warning(
+                "Datadog-opentracing requires C++14. Your compiler is unknown. Assuming it supports C++14."
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "4.0":
+            tc.variables["CMAKE_CXX_FLAGS"] = "-ftemplate-depth=1024"
+        tc.variables["BUILD_PLUGIN"] = False
+        tc.variables["BUILD_SHARED"] = self.options.shared
+        tc.variables["BUILD_STATIC"] = not self.options.shared
+        tc.variables["BUILD_TESTING"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_PLUGIN"] = False
-        self._cmake.definitions["BUILD_SHARED"] = self.options.shared
-        self._cmake.definitions["BUILD_STATIC"] = not self.options.shared
-        self._cmake.definitions["BUILD_TESTING"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
 
     def package_info(self):
-        # TODO: back to global scope in conan v2 once cmake_find_package_* generators removed
-        self.cpp_info.components["dd_opentracing"].libs = ["dd_opentracing"]
-        self.cpp_info.components["dd_opentracing"].defines.append(
-            "DD_OPENTRACING_SHARED" if self.options.shared else "DD_OPENTRACING_STATIC"
-        )
+        self.cpp_info.libs = ["dd_opentracing"]
+        self.cpp_info.defines.append("DD_OPENTRACING_SHARED" if self.options.shared else "DD_OPENTRACING_STATIC")
         if self.settings.os in ("Linux", "FreeBSD"):
-            self.cpp_info.components["dd_opentracing"].system_libs.append("pthread")
-
-        # TODO: to remove in conan v2 once cmake_find_package_* generators removed.
-        #       Do not support these names in CMakeDeps, it was a mistake, upstream doesn't export targets
-        self.cpp_info.names["cmake_find_package"] = "DataDogOpenTracing"
```

Comment:
I see no new properties, but removing old components. Was it wrong before?

Reply:
I based it on the comment above:
```
# TODO: to remove in conan v2 once cmake_find_package_* generators removed.
#       Do not support these names in CMakeDeps, it was a mistake, upstream doesn't export targets
```